### ISSUE TITLE
Extension tab is closed and marketplace search lost upon reload

### ISSUE THREAD

**Issue by PunchyRascal** [Microsoft/vscode#18970]
- VSCode Version: Code 1.8.1 (ee428b0eead68bf0fb99ab5fdc4439be227b6281, 2016-12-19T14:49:23.350Z)
- OS Version: Windows_NT ia32 10.0.14393

Steps to Reproduce:
1. search for extension, install it
2. you are forced to reload the window to see if the extension does what you want
3. upon reload the tab with extension details is not restored and marketplace panel lost the serach query, so you are forced to start your search from the beginning

In other words, finding a working extension for a given task (e.g. formatting an XML file) can be very frustrating since many extensions don't work at all or don't do what they say they do.

**Comment by PunchyRascal** [comment]
Also when you find out the extension doesn't work, you actually have to look it up among all existing extensions in the marketplace to uninstall it (apart from scrolling through all of your installed extensions, of which there may be many). That seems very bizarre. The fact, that themes are counted among extensions adds hugely to this problem.

**Comment by joaomoreno** [comment]
For first comment: #5675

Can you please file another issue for the second comment?
